eclipse-collections-code-generator/src/main/resources/impl/list/immutable/immutablePrimitiveSingletonList.stg

```diff
@@ -456,6 +458,12 @@ final class Immutable<name>SingletonList implements Immutable<name>List, Seriali
         }
         return Lists.immutable.with(PrimitiveTuples.pair(this.element1, Iterate.getFirst(iterable)));
     }
+
+    @Override
+    public Mutable<name>Stack toStack()
+    {
+        return <name>Stacks.mutable.withAll(this);
```

Comment:
This can be simplified to just add the single element to the stack as the `toList` and `toSet` methods do.

Reply:
Agreed.